vars/groups.groovy

```diff
@@ -0,0 +1,108 @@
+import groovy.json.JsonSlurper
+import groovy.json.JsonOutput
+
+def createGroup(String groupName, List groupMembers) {
+   return JsonOutput.toJson([
+      name: groupName,
+      members: groupMembers
+   ]);
+}
+
+def getGroupMembers(groups, String groupName) {
+    for(group in groups) {
+        if(group.name == groupName) {
+            return group.members;
+        }
+    }
+
+    log.error("Tried to get group members for group '${groupName}', but this group couldn't be found!");
+    return [];
+}
+
+def getGroupType(groups, String groupName) {
+    for(group in groups) {
+        if(group.name == groupName) {
+            return group.type;
+        }
+    }
+
+    log.error("Tried to get group type for group '${groupName}', but this group couldn't be found!");
+    return [];
+}
+
+def mentionGroup(groups, String groupName) {
+    def groupMembers = getGroupMembers(groups, groupName);
+    def groupType = getGroupType(groups, groupName);
+
+    if(!groupMembers && !groupType) {
+        return '';
+    }
+
+    def tagTemplate = '';
+
+    switch(groupType.toLower()) {
+        case 'user':
+            tagTemplate = '<@{id}>';
+            break;
+        case 'role':
+            tagTemplate = '<@&${id}>';
+            break;
+        case 'channel':
+            tagTemplate = '<#{id}>';
```

Comment:
Forgot $ here ( <#${id}> )

Reply:
Ironically, it's the opposite; I forgot to remove it in the role tag.
The reason it's not needed here is because it's used as a template, and `id` is not meant to be substituted at this point in the function. I'll replace the `{}` with `[]` so it's more obvious.